### PR TITLE
SConstruct: init git submodules if necessary

### PR DESCRIPTION
**Pull request by rmsr**
Pretty simple change, but maybe there's more I should add?

### CODE REVIEW

File: code/SConstruct [hunk 1]
```diff
@@ -73,6 +73,18 @@ opt_olevel =         get_option('olevel',      '')
 opt_cxx =            get_option('CXX',         'g++')
 
 
+## ensure submodules are ready to go
+submodules = subprocess.check_output(['git','submodule','status']).split()
+if any(sub.startswith('-') for sub in submodules):
+    print >>sys.stderr, 'Initializing submodule dependencies, please wait...'
```
Comment (cizra) [inline code/SConstruct]:
```py
from __future__ import print_function
import sys

print('lala', file=sys.stderr)
```

By the way, why do you consider this comment to be an error?

File: code/SConstruct [hunk 2]
```diff
@@ -73,6 +73,18 @@ opt_olevel =         get_option('olevel',      '')
...
+    try:
+        subprocess.check_call(['git','submodule','init'])
+        subprocess.check_call(['git','submodule','update'])
```
Comment (cizra) [inline code/SConstruct]: `git submodule update --init`?

File: code/SConstruct [hunk 3]
```diff
@@ -73,6 +73,18 @@ opt_olevel =         get_option('olevel',      '')
...
+    except submodule.CalledProcessError, e:
```
Comment (cizra) [inline code/SConstruct]:
```py
except Something as e:
```

File: code/SConstruct (hunk 2 above)
Comment (rmsr) [reply code/SConstruct]: ooh nice flag, combined

File: code/SConstruct (hunk 3 above)
Comment (rmsr) [reply code/SConstruct]: I don't use e anyway, removed

**Comment by rmsr** [comment]
Addressed the good feedback, merging

**Pull request merged**